src/nycdb/plugins.py

```diff
@@ -0,0 +1,22 @@
+from pathlib import Path
+import importlib
+from importlib.util import find_spec
+import pkgutil
+
+
+class Plugin:
+    def __init__(self, name: str):
+        self.name = name
+        self.spec = find_spec(name)
+        self.root_dir = Path(self.spec.origin).parent.resolve()
+
+    def import_submodule(self, name: str):
+        return importlib.import_module(f"{self.name}.{name}")
+
+
+def iter_plugins():
+    yield Plugin('nycdb')
+
+    for finder, name, ispkg in pkgutil.iter_modules():
+        if ispkg and name.startswith('nycdb_'):
```

Comment:
take this comment with these caveats -- i'm not  a python packaging expert or a security expert.

does give open up some security concerns? What's stopping someone from creating package called nycdb_evil_plugin, uploading it to pypi and having it get installed automatically here?

Reply:
Oh, this will only activate the plugin if it's locally installed on the user's system, not if it's available on PyPI.  As with any package (since packages can execute arbitrary code when they're installed), a developer is expected to make sure they're not installing anything malicious when they `pip install` it.  But once a package is on a user's system it's assumed to not be malicious.

Also, given that this [convention](https://packaging.python.org/guides/creating-and-discovering-plugins/#using-naming-convention) is recommended by the official Python documentation and used by Flask (a package that takes security very seriously) I think we should be good!